src/forms/datepicker/DatePicker.tsx

```diff
@@ -0,0 +1,423 @@
+                                                                                            ? 0
+                                                                                            : -1
+                                                                                        : isToday
+                                                                                        ? 0
+                                                                                        : -1
+                                                                                }
+                                                                                autoFocus={
+                                                                                    selectedSameMonth
+                                                                                        ? isSelected
+                                                                                            ? true
+                                                                                            : false
+                                                                                        : isToday
+                                                                                        ? true
+                                                                                        : false
+                                                                                }
+                                                                                onClick={this.handleSelectedDate.bind(
+                                                                                    this,
+                                                                                    calendar.toDate(),
+                                                                                )}
+                                                                            >
+                                                                                {calendar.date()}
+                                                                            </button>
+                                                                        </div>
+                                                                    </td>
+                                                                );
+                                                                calendar.add(1, "day");
+                                                                return day;
+                                                            })}
+                                                        </tr>
+                                                    );
+                                                })}
+                                            </tbody>
+                                        </table>
+                                    </div>
+                                )}
+                                {viewMode === ViewMode.Month && (
```

Comment:
You need to render different views for different viewMode let's split them on components/render functions

Reply:
Can you explain in more detail please?

Reply:
We will keep components that should be rendered for specific viewMode separately. For example: we move whole <div className="monthpicker"> to another component? As a result we will render the component.
What do you think? Do we need keep components smaller?

Reply:
@alexander-rassanov I think for now it would be okay to keep the code together in a single component.  Maybe when time allows, we can do some refactoring.  Because `clarity-react` is a toolkit to be consumed by other applications, the Datepicker component will still be easily consumed/used without this change.  For now, lets keep the same if you're okay with that.